### PR TITLE
Initial Tensorboard support

### PR DESCRIPTION
**Pull request by msaintja**
Do not merge yet:
- [x] Support for DSN
- [x] Support for Sumgan

Optional:
- [x] Hyperparameters logging support
- [ ] Possibly other metrics
- [ ] Log images/videos?
- [ ] Log per-video loss/F-score as well?

Launch tensorboard with `tensorboard --logdir=logs` in root dir (i.e. `summarizer`).
Make sure you have tensorboard installed in the first place (`pip install tensorboard`).

By default, there will be a plot with CV F-scores for each model & each dataset. If you want your list of runs to be a bit less crowded, you can use this regex: `^[^\\]*$`

### CODE REVIEW

**Comment by msaintja** [comment]
Reference: https://pytorch.org/docs/stable/tensorboard.html

File: summarizer/main.py [hunk 1]
```diff
@@ -10,8 +11,16 @@ def train(hps):
...
     for splits_file in hps.splits_files:
         print("Start training on {}".format(splits_file))
+
+        hps.current_dataset = splits_file
+        # Cosmetic for Tensorboard
+        if "splits/" in hps.current_dataset:
+            hps.current_dataset = hps.current_dataset.split("splits/")[1]
+        if "_splits" in hps.current_dataset:
+            hps.current_dataset = hps.current_dataset.split("_splits")[0]
+
```
Comment (sylvainma) [inline summarizer/main.py:21]: We may move this into utils/config.py directly as there is already code related to that

File: summarizer/main.py [hunk 2]
```diff
@@ -95,3 +109,6 @@ def test(hps):
         test(hps)
     else:
         train(hps)
+
+    # Close the Tensorboard writer
+    hps.writer.close()
```
Comment (sylvainma) [inline summarizer/main.py:114]: Suggestion: transfer this into config class within the `def __del__(self):`

Comment (sylvainma) [reply summarizer/main.py:114]: Update: I tried but it failed, when `self.writer.close()` is called in the `__del__(self)` it no longer works (wait infinitely). After further investigation, the `.close()` on the writers in the loop [here](https://github.com/pytorch/pytorch/blob/3e3498cf03b6ea7becf34597eea7ff350d718f95/torch/utils/tensorboard/writer.py#L1019) fails

Comment (msaintja) [reply summarizer/main.py:114]: This needs more investigation. I see some workarounds here: https://github.com/facebookresearch/detectron2/issues/121

Comment (msaintja) [reply summarizer/main.py:107]: It seems like workarounds for this revolve around external/calling code closing the writer. While not optimal, this is how we have it implemented currently as well.
